aiutils/tftools/train.py

```diff
@@ -0,0 +1,104 @@
+""" This module defines a hepler class for multiple rate optimizers in Tensorflow
+
+The MultiRateOptimizer class provides a slightly simpler version of the tf.train.Optimizer API
+ - compute_gradients(loss)
+ - apply_gradients(grads_and_vars, global_step)
+ - minimize(loss, global_step)
+
+To add variables and corresponding optimizers use
+ - add_variables(variables, optimizer)
+
+Usage:
+    a = tf.Variable(1)
+    b = tf.Variable(2)
+    c = tf.Variable(3)
+
+    loss = f(a,b,c)
+
+    optimizer = MultiRateOptimizer()
+    optimizer.add_variables([a,b], tf.train.GradientDescentOptimizer(.1))
+    optimizer.add_variables([c], tf.train.GradientDescentOptimizer(.01))
+
+    min_opt = optimizer.minimize(loss)
+
+    min_opt.eval()
+"""
+
+import tensorflow as tf
+import itertools
+
+
+class MultiRateOptimizer():
+    """ Class for managing a multi-rate optimization problem """
+
+    def __init__(self):
+        self.optimizers = []
+        self.variables = []
+
+    def add_variables(self, variables, optimizer):
```

Comment:
If the only intended purpose is to apply different learning rate to different variables but still use same optimizer such as Adam/SGD etc, then it might make sense to set the optimizer once and make `add_variables` take `variables` and 'learning_rate' as arguments instead of `optimizer`